### PR TITLE
[SPARK-37235][PYTHON] Inline type hints for python/pyspark/mllib/stat/distribution.py and __init__.py

### PR DESCRIPTION
**Pull request by dchvn**
### What changes were proposed in this pull request?
Inline type hints for python/pyspark/mllib/stat/distribution.py and __init__.py

### Why are the changes needed?
We can take advantage of static type checking within the functions by inlining the type hints.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing Tests

### CODE REVIEW

File: python/pyspark/mllib/stat/distribution.py [hunk 1]
```diff
@@ -15,12 +15,14 @@
...
-
 __all__ = ['MultivariateGaussian']
 
+from typing import NamedTuple
+
+from pyspark.mllib.linalg import Matrix, Vector
+
 
-class MultivariateGaussian(namedtuple('MultivariateGaussian', ['mu', 'sigma'])):
+class MultivariateGaussian(NamedTuple):
```
Comment (HyukjinKwon) [inline python/pyspark/mllib/stat/distribution.py:25]: Hm, is't it a breaking change? Previously it inherited an actual named tuple that works like named tuple. Now it inherits the typing class of `NamedTuple`.

**Comment by HyukjinKwon** [comment]
Hey, @dchvn, it would be great to double check and update your old PRs before opening new ones. Many of them are missing details like that and it takes more time to review ...

**Comment by HyukjinKwon** [comment]
Also it would be much easier to review if you add some justification on `ignore`s or unusual code changes.

File: python/pyspark/mllib/stat/distribution.py (hunk 1 above)
Comment (dchvn) [reply python/pyspark/mllib/stat/distribution.py:25]: Please take a look at https://github.com/apache/spark/pull/34375#discussion_r738143060 .
I followed that one. Are they different?

Comment (HyukjinKwon) [reply python/pyspark/mllib/stat/distribution.py:25]: Oh, okay, it was my bad. I didn't reliased that the typing annotation can be same https://docs.python.org/3/library/typing.html#typing.NamedTuple. Resolving this.

File: python/pyspark/mllib/stat/__init__.pyi [hunk 2]
```diff
@@ -1,29 +0,0 @@
...
-    KernelDensity as KernelDensity,
-)
-from pyspark.mllib.stat._statistics import (  # noqa: F401
-    MultivariateStatisticalSummary as MultivariateStatisticalSummary,
-    Statistics as Statistics,
-)
-from pyspark.mllib.stat.distribution import (  # noqa: F401
-    MultivariateGaussian as MultivariateGaussian,
-)
-from pyspark.mllib.stat.test import ChiSqTestResult as ChiSqTestResult  # noqa: F401
```
Comment (HyukjinKwon) [inline python/pyspark/mllib/stat/__init__.pyi:29]: Is this safe to remove, @zero323?

**Review by HyukjinKwon — COMMENTED** [review]
Looks good otherwise.

File: python/pyspark/mllib/stat/__init__.pyi (hunk 2 above)
Comment (zero323) [reply python/pyspark/mllib/stat/__init__.pyi:29]: Seem like it @HyukjinKwon. There is nothing here, that we cannot pick out of `py` file.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dchvn** [comment]
> Hey, @dchvn, it would be great to double check and update your old PRs before opening new ones. Many of them are missing details like that and it takes more time to review ...


Please see my old PRs and give me your opinions! I have updated with reviewers' comments! Many thanks 😄

**Pull request closed without merging**